larva/src/main/java/org/frankframework/larva/TestRunStatus.java

```diff
@@ -0,0 +1,288 @@
+			String description = appConstants.getProperty("scenariosroot" + j + ".description");
+			String m2eFileName = appConstants.getProperty("scenariosroot" + j + ".m2e.pom.properties");
+
+			if (directory == null) {
+				break;
+			}
+			if (description == null) {
+				out.errorMessage("Could not find description for directory scenariosroot%d [%s]".formatted(j, directory));
+				continue;
+			}
+			if (scenariosRoots.containsKey(description)) {
+				out.errorMessage("A root directory named [%s] already exists".formatted(description));
+				continue;
+			}
+			String parent = determineParentPath(realPath, m2eFileName);
+			directory = LarvaUtil.getAbsolutePath(parent, directory, true);
+			if (scenariosRoots.containsValue(directory)) {
+				out.errorMessage("A root directory with path [%s] already exists".formatted(directory));
+				continue;
+			}
+			if (new File(directory).exists()) {
+				out.debugMessage("directory for [" + description + "] exists: " + directory);
+				scenariosRoots.put(description, directory);
+			} else {
+				out.debugMessage("directory [" + directory + "] for [" + description + "] does not exist, parent [" + parent + "]");
+				scenariosRoots.put("X " + description, directory);
+			}
+		}
+		out.debugMessage("Read scenariosrootdirectory parameter");
+		out.debugMessage("Get current scenarios root directory");
+		if (StringUtils.isEmpty(larvaConfig.getActiveScenariosDirectory())) {
+			String currentScenariosRootDirectory;
+			String scenariosRootDefault = appConstants.getProperty("scenariosroot.default");
+			if (StringUtils.isNotEmpty(scenariosRootDefault) && scenariosRoots.containsKey(scenariosRootDefault)) {
+				currentScenariosRootDirectory = scenariosRoots.get(scenariosRootDefault);
+			} else if (!scenariosRoots.isEmpty()) {
+				currentScenariosRootDirectory = scenariosRoots.get(scenariosRoots.firstKey());
+			} else {
+				currentScenariosRootDirectory = null;
+			}
+			larvaConfig.setActiveScenariosDirectory(currentScenariosRootDirectory);
+		}
+		scenarioDirectories = scenariosRoots;
+		return larvaConfig.getActiveScenariosDirectory();
+	}
+
+	private String determineParentPath(String realPath, String m2eFileName) {
+		String parent = realPath;
+		if (m2eFileName != null) {
+			File m2eFile = new File(realPath, m2eFileName);
+			if (m2eFile.exists()) {
+				out.debugMessage("Read m2e pom.properties: " + m2eFileName);
+				Properties m2eProperties = LarvaUtil.readProperties(out, m2eFile);
+				parent = m2eProperties.getProperty("m2e.projectLocation");
+				out.debugMessage("Use m2e parent: " + parent);
+			}
+		}
+		return parent;
+	}
+
+	public void readScenarioFiles(ScenarioLoader scenarioLoader) {
+		allScenarios = scenarioLoader.readScenarioFiles(larvaConfig.getActiveScenariosDirectory());
+	}
+
+	public List<Scenario> getScenariosToRun(@Nonnull String execute) {
+
+		List<Scenario> scenarios;
+		if (execute.endsWith(".properties")) {
+			out.debugMessage("Read one scenario");
```

Comment:
Ik neig nog heel erg naar log4j2.... Misschien een mooie volgende stap 😄

Reply:
Log4J is niet direct zichtbaar voor de gebruiker. We moeten eens goed kijken welke logging we voor de gebruiker zichtbaar willen maken.
Maar dit is er eentje waar de gebruiker eigenlijk echt helemaal niks aan heeft, zeker niet als er 500 scenarios worden ingelezen bij het laden van de Larva pagina. 😬